### PR TITLE
Provide jenkins snapshots only from master branch

### PR DESCRIPTION
**Pull request by carstene1ns**
Now uses static link approach, like mentioned.
There are some TODOs left.

Related: #49.

OLD Content:
---

Works so far.

This currently does 12 request to the jenkins api and takes roughly a second.
I am thinking of disabling this for PR builds and only applying when run in "release mode" (e.g. from systemd timer).

### CODE REVIEW

**Comment by carstene1ns** [comment]
Meh.
- macos build is too old (artifacts already removed)
- linux build has "wrong" version (this slipped through in the 0.6.2.1 pr)

**Comment by Ghabry** [comment]
You suggested to just copy the artifacts somewhere when the build is on the master branch and just link them. I think this is the better solution. But we still have some time now to think about this because there is no new stable release soon.

**Comment by carstene1ns** [comment]
Problem is here that the individual agents cannot access the master server, jenkins will upload artifacts through its own protocol. So this either needs yet another server cronjob to collect them from the workspace (with parsing involved) or external scripting.
I have added some copy commands now and collect files here: https://ci.easyrpg.org/downloads/
for external agents this involves having some ssh key to be able to `scp` the artifacts.
On master, this involved giving jenkins user ssh access. This should be secure as we only allow keys anyway. As additional security the shell has been changed to `scponly`, to not allow interactive terminal sessions.

**linux/android**
```sh
# provide download
test "x$GIT_BRANCH" = "xorigin/master" && \
  cp -rup easyrpg-player-static.tar.gz $JENKINS_HOME/ci-archives/
```

**3ds/switch/wii**
```sh
# provide download
test "x$GIT_BRANCH" = "xorigin/master" || exit 0

cd builds/3ds
scp -p easyrpg-player-3ds.zip easyrpg-player.cia easyrpg-player.elf.zip easyrpg.org:ci-archives/3ds/
```

---

About parsing jenkins stuff, it is even more ugly than using the api:
```
jenkins@easyrpg ~/j/player-switch> exa -aT
.
├── builds
│  ├── 522
│  │  ├── archive
│  │  │  └── builds
│  │  │     └── switch
│  │  │        ├── easyrpg-player-switch.zip
│  │  │        ├── easyrpg-player.elf
│  │  │        └── easyrpg-player.nro
│  │  ├── build.xml                                      <- parse xml with loop to get outer
│  │  ├── changelog.xml                                     tag from inner build number
│  │  ├── changelog.xml.temp2
│  │  └── log
│  ├── [some other builds with same content]
│  ├── legacyIds
│  └── permalinks                                        <- parse `lastStableBuild 522`
├── config.xml
└── nextBuildNumber
```

**Review by fdelapena — APPROVED** [review]

**Comment by carstene1ns** [comment]
Finally ready now. See https://deploy-preview-50--easyrpg.netlify.app/player/downloads/#snapshot-linux

File: content/player/downloads.html [hunk 1]
```diff
@@ -202,66 +212,72 @@ <h3>EasyRPG Player continuous builds</h3>
...
+					<a href="https://ci.easyrpg.org/downloads/android/EasyRpgPlayerActivity.apk">Locally hosted App (.apk)</a>
 					<br>
 					Installing from "unknown sources" must be activated in the Android settings.
 				</p>
 			</div>
 			<div id="snapshot-libretro">
 				<h4>libretro</h4>
 				<p>
-					<a href="<%= jenkins_link("player-win32", "bin/easyrpg_libretro.zip") %>">Windows 32 bit core</a>, <a href="<%= jenkins_link("player-win64", "bin/easyrpg_libretro_x64.zip") %>">Windows 64 bit core</a>
+					<a href="https://ci.easyrpg.org/downloads/libretro/easyrpg_libretro_win32.zip") %>">Windows 32 bit core</a>, <a href="https://ci.easyrpg.org/downloads/libretro/easyrpg_libretro_win64.zip") %>">Windows 64 bit core</a>
```
Comment (Ghabry) [inline content/player/downloads.html:252]: The ``%>`` must be removed here (twice)

Comment (carstene1ns) [reply content/player/downloads.html:252]: whoops

**Pull request merged**